### ISSUE TITLE
Supporting jemalloc memory allocator

### ISSUE THREAD

**Issue by jetchirag** [pwndbg/pwndbg#2174]
Opening this issue to discuss and keep track of development for jemalloc support in pwndbg. This project is a part of my GSoC 24 proposal.

Original Details of the project: https://github.com/pwndbg/pwndbg/issues/2014#issuecomment-1931092609

Currently I am working on the first PR to print basic extent info, got delayed due to a health issue. Will post my current dev environment in case anyone wants to catch up along with all the resources I found useful.

**Comment by jetchirag** [comment]
I'm using vscode remote development on a VPS. 
- pwndbg 
- [jemalloc](https://github.com/jemalloc/jemalloc/blob/dev/INSTALL.md)  (5.3.0)

run.sh
```
cd "$(dirname "$0")"

filename=test

rm -f $filename
gcc -g -O0 $filename.cpp -o $filename `jemalloc-config --libdir`/libjemalloc.a `jemalloc-config --libs`

#  if argument y is passed if n then run simple else nothing
if [ "$1" == "y" ]; then
    gdb -ex run --args $filename
elif [ "$1" == "n" ]; then
    ./$filename
fi

```

Sample test file:
```
#include <stdlib.h>
#include <iostream>
#include <jemalloc/jemalloc.h>

void breakhere(){}

int main(void)
{

    char *block;
    for(int i=0; i<(4096/8); i++){
        block = (char *)malloc(8);
        for (int i = 0; i < 8; i++){
            block[i] = 'A' + (i % 26);
        }
        // breakhere();
    }

    breakhere();

    return 0;
}
```

**Comment by CptGibbon** [comment]
Looks like you're off to a great start 👍
Don't sweat the small delay, we're technically still in the community bonding period.
If you open a draft PR with your basic setup (jemalloc installed via `setup-dev.sh`, test binary etc.) I can clone that so we're on the same page.

**Comment by disconnect3d** [comment]
Basic JEMalloc summport was implemented in #2176. Shall we close this now @CptGibbon or do we want more things to be implemented for jemalloc?

**Comment by CptGibbon** [comment]
> Basic JEMalloc summport was implemented in #2176. Shall we close this now @CptGibbon or do we want more things to be implemented for jemalloc?

Perhaps best to close this now and any more specific feature requests can be brought up in future.
